IOTConnectSDK/app/src/main/java/com/softweb/iotconnectsdk/activity/GatewayChildDevicesActivity.kt

```diff
@@ -0,0 +1,144 @@
+package com.softweb.iotconnectsdk.activity
+
+import android.os.Bundle
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.iotconnectsdk.interfaces.DeviceCallback
+import com.softweb.iotconnectsdk.R
+import com.softweb.iotconnectsdk.activity.FirmwareActivity.sdkClient
+import kotlinx.android.synthetic.main.activity_gateway_child_devices.btnCreateDevice
+import kotlinx.android.synthetic.main.activity_gateway_child_devices.btnDeleteDevice
+import kotlinx.android.synthetic.main.activity_gateway_child_devices.etDisplayName
+import kotlinx.android.synthetic.main.activity_gateway_child_devices.etUniqueId
+import kotlinx.android.synthetic.main.activity_gateway_child_devices.spTags
+import kotlinx.android.synthetic.main.activity_gateway_child_devices.toolbar
+import org.json.JSONObject
+
+class GatewayChildDevicesActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener,
+    DeviceCallback {
+
+    private var tagsList: ArrayList<String>? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_gateway_child_devices)
+
+        tagsList = intent.extras?.getStringArrayList("tagsList")
+
+        // calling the action bar
+
+        // showing the back button in action bar
+        setSupportActionBar(toolbar)
+        supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_back_arrow);
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+
+        val adapter = ArrayAdapter(
+            this,
+            android.R.layout.simple_spinner_dropdown_item, tagsList!!
+        )
+        spTags.adapter = adapter
+        spTags.onItemSelectedListener = this
+
+        btnCreateDevice.setOnClickListener {
+            if (checkValidation()) {
+
+                sdkClient.createChildDevice(etUniqueId.text.toString(),spTags.selectedItem.toString(),etDisplayName.text.toString())
+            }
+        }
+
+        btnDeleteDevice.setOnClickListener {
+            if (etUniqueId.text.toString().isEmpty()) {
+                Toast.makeText(
+                    this@GatewayChildDevicesActivity,
+                    getString(R.string.alert_enter_unique_id),
+                    Toast.LENGTH_SHORT
+                ).show()
+            } else {
+
+                sdkClient.deleteChildDevice(etUniqueId.text.toString())
+            }
+
+        }
+    }
+
+    override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+
+    }
+
+    override fun onNothingSelected(parent: AdapterView<*>?) {
+
+    }
+
+    override fun onDeviceCommand(message: String?) {
+        TODO("Not yet implemented")
```

Comment:
@sw-tt-rajkiranvank @sw-tt-tusharojha this functionality is not in scope?

Reply:
HI @sw-ot-deveshmevada 
I have put comment so user can understand that